libs/datamodel/core/src/configuration/datasource.rs

```diff
@@ -1,7 +1,13 @@
 use crate::configuration::preview_features::PreviewFeatures;
+use crate::diagnostics::DatamodelWarning;
 use datamodel_connector::{Connector, ConnectorCapabilities};
 use serde::Serialize;
 
+pub struct ValidatedDatasource {
+    pub datasource: Datasource,
+    pub warnings: Vec<DatamodelWarning>,
+}
```

Comment:
We could consider turning this pattern into a generic struct:
```
pub struct ValidatedThing<T> {
  pub thing: T,
  pub warnings: Vec<DatamodelWarning>,
}
pub type ValidatedDatasource = ValidatedThing<Datasource>
```

But don't know if it's worth it. That probably depends on the feedback of people who use it.

Reply:
I think that's good to add here, makes it easier to add more ValidatedThings in the future.